configure.ac

```diff
@@ -451,6 +451,7 @@ AC_DEFUN([LIBATOMIC_CHECKER],[
 #include <cstdint>
       int
       main(int, char **) {
+          int foo = __atomic_exchange_1(&foo, 2, __ATOMIC_ACQ_REL);
```

Comment:
```
$ git grep __atomic_exchange_1 | wc -l
0
```

We should not add non-standard function calls to this libatomic checker because they will break the build on platforms that lack those specific calls (not to mention causing reader confusion from seeing seemingly unused by Squid code calls used for testing the necessity of linking with this standard library).

To fix this build problem, we should figure out what standard std::atomics API requires __atomic_exchange_1() in your environment and then use that API in this checker. I have not checked myself, but I bet that standard API is compare_exchange_strong(). Could you please try testing for that instead?

```
$ git grep compare_exchange
src/ipc/StoreMap.cc:    return s.waitingToBeFreed.compare_exchange_strong(expected, true);
src/ipc/mem/PageStack.cc:    } while (!node.compare_exchange_weak(oldValue, newValue.pack()));
src/ipc/mem/PageStack.cc:    } while (!node.compare_exchange_weak(oldValue, newValue));
```